justifi-tech/web-component-library
# New Component: TokenizePaymentMethod

The `TokenizePaymentMethod` web component will be used for our customers with subscription style use cases as we will be deprecating `PaymentForm` in version `5.0.0`. `TokenizePaymentMethod` almost entirely re-uses code from `Checkout`, has a more appropriate name, and styling can be applied to it in a way that is consistent with `Checkout`.

Developer considerations
--------------

- On any task
- On new features
- On bugs


Developer QA steps
--------------------

Pull the example from https://github.com/justifi-tech/web-component-library/pull/757

- Run `npm start:tokenize-payment-method`
- Click the "submit" button without entering any info. It should show inline errors.
- Fill out the fields using valid test card data and click "submit" again. The output pane in the example should show the created token

External submit button
- Modify the code in the example to:
  - Hide the submit button `hide-submit-button="true"`
  - Create an external button with an id
  - Create a click handler for the external button that calls `tokenizePaymentMethod`
  - click the button to tokenize, and the token should be output to the pane just as it did in the original example
